src/bun.js/webcore/response.zig

```diff
@@ -2141,24 +2142,55 @@ pub const Fetch = struct {
             const PercentEncoding = @import("../../url.zig").PercentEncoding;
             var path_buf2: [bun.MAX_PATH_BYTES]u8 = undefined;
             var stream = std.io.fixedBufferStream(&path_buf2);
-            const url_path_decoded = path_buf2[0 .. PercentEncoding.decode(
+            var url_path_decoded = path_buf2[0 .. PercentEncoding.decode(
                 @TypeOf(&stream.writer()),
                 &stream.writer(),
                 url.path,
             ) catch {
                 globalThis.throwOutOfMemory();
                 return .zero;
             }];
```

Comment:
It looks this `decode` function returns `error.DecodingError` and not `error.OutOfMemory` so it might be better to capture the error value and do: `globalThis.throwError(err, ...)`

Reply:
Yeah makes much more sense fixed here: https://github.com/oven-sh/bun/commit/36ddac0b644b1087e76cac906748fb283a552a7d